flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/files/StaticFileServerHandler.java

```diff
@@ -234,8 +237,17 @@ private void respondAsLeader(ChannelHandlerContext ctx, HttpRequest request, Str
 							if (!rootURI.relativize(requestedURI).equals(requestedURI)) {
 								logger.debug("Loading missing file from classloader: {}", requestPath);
 								// ensure that directory to file exists.
-								file.getParentFile().mkdirs();
-								Files.copy(resourceStream, file.toPath());
+								if (!file.getParentFile().mkdirs()) {
+									throw new IOException("Could not create directories for file " + file);
+								}
+								synchronized (COPY_LOCK) {
```

Comment:
There is nothing stopping us from changing to eager loading later on, so I opted for the fastest, yet still reasonable, fix for the problem.